baseplate/frameworks/thrift/__init__.py

```diff
@@ -82,7 +82,10 @@ def _set_remote_context(self, request_context: RequestContext) -> Iterator[None]
             header_dict = {}
             for k, v in headers.items():
                 try:
-                    header_dict[k.decode()] = v.decode()
+                    # this is only for w3c trace headers
+                    key = k.decode()
+                    if key in ["traceparent", "tracestate", "Trace", "Span", "Sampled", "Flags"]:
+                        header_dict[key] = v.decode()
                 except UnicodeDecodeError:
                     self.logger.info(f"Unable to decode header {k.decode()}, ignoring.")
```

Comment:
Another possibility if these can contain unsanitized user input would be to ignore encoding errors, e.g.

```python
header_dict[k.decode(errors="replace")] = v.decode(errors="replace")
```

That would replace invalid UTF-8 sequences with �.

Not sure if that is appropriate here or not. I wouldn't want to hide legitimate errors like this, but if these are raw bytes from users, then we're always going to get some quantity of malformed requests from poorly-behaving UAs and we wouldn't want to hide real errors among decode errors that we can't actually do anything about.

Reply:
For the most part we're building this dictionary so that we can use the otel propagator injection code which expects a dictionary of strings. We' don't really care about any of the other headers which is why I shifted towards making it more specific.

Reply:
Invalid values should be safe to just drop, if we do encode them with � it would just be considered invalid by otel later.